### ISSUE TITLE
Code cleanup

### ISSUE THREAD

**Issue by bkazemi** [scanmem/scanmem#181]
`scanmem` code doesn't seem to have a consistent style. For example, certain line lengths are quite long. Maybe in the future we could restrict all lines to 80 chars.

**Comment by 12345ieee** [comment]
I guess #177 falls under this, too.

**Comment by 12345ieee** [comment]
Another point, in the codebase there are things like:

```
            while(((*string_parameter) == ' ')
                 ||(*string_parameter) == '\t') 
```

which would be much better changed to an [isspace()](http://www.cplusplus.com/reference/cctype/isspace/) call.

**Comment by bkazemi** [comment]
I think this should be first on the list for v0.17 to reduce diffs by ensuring newly added code conforms to our style requirements. I'm going to write up my comprehensive proposal for the `libscanmem` style here and @sriemer and the others can look over it and change what they wish so we can come to a consensus. Since I'm the one nagging here and the changes are quite burdensome, I'll take up the task myself if others don't have the time.

**Comment by bkazemi** [comment]
General
----------
* Max 80 character line length.
* Prefer functions, macros, etc. in this order:
 1. POSIX
 2. Standard libc
 3. OS specific with proper wrapping
 4. Custom

Variable names
--------------------
#### Changes
* Shorten common names to their well-known condensed forms
 * array => arr
 * address => addr
 * length => len
 * string => str
 * index => idx
 * number => num
 * element => elem
 * value(s) => val(s)
 * {max,min}imum => max, min
* Try not to add unnecessary words to a variable name

Comments
--------------
* Keep all multi-line comments in this form:
```
/*
 * line 1
 * line 2
 */
```
* Function header comments in multi-line format, with no newline between end of comment and start of function declaration. Example:
```
/*
 * This function foo's bar.
 */
int foo(bar *b) ...
```
* Always use `/**/` type comments when a comment is on it's own line (_i.e,_ nothing else is on that line)
* Use `//` type comments at the end of a line to add a clarifying comment when it seems more natural that adding it at the line before the statement. Example:
```
if (a > b) { // bad number
```
* Keep consecutive comments connected in the form described in the first bullet point. Example:
```
/* Comment 1 */
/* Comment 2 on next line */

/*
 * Comment 1
 * Comment 2 (maybe add space if comments are unrelated)
 */
```

Preprocessor
-----------------
* Add one space for every level of nesting in preprocessor conditionals. Example:
```
#ifdef a
# include <header.h>
#else
# warning no header.h!
#endif

#if lvl1
# if lvl2
#  include "abc.h"
# endif
#endif
```

This is a _general_ outline proposal, I realize there are always exceptions to the rule. This is unfinished, I'll come back and add things, in the meantime, any input is welcome!

**Comment by 12345ieee** [comment]
I am an absolute amateur, so take everything I say as the layman's opinion.

These rules seem incredibly restrictive. Do professional environments have guidelines so strict?

This will both put an immense review burden on @bkazemi and discourage people from contributing.
I get that scanmem (not GC) is a mature codebase that needs top-notch contributions only, but we're not exactly swimming in contributions here (quite the contrary) and I don't feel the need to weed out people.

**Comment by bkazemi** [comment]
Actually, compared to most big projects, this guideline is comparatively lax. As I've said, this is only an _outline_; it helps code readability and keeps things clean. As for new contributions, it is actually fairly easy to conform; All you have to do is match your style to what's already there. I wrote this proposal for existing contributors in mind, so we can discuss what's best.

**Comment by bkazemi** [comment]
I'll be absent for the better part of two months, so I guess this has to go on the back-burner.

**Comment by 12345ieee** [comment]
@bkazemi , any news on this?

I have a rather big round of commits incoming, if you have already done part of this, we can merge it before mine, and I can adapt what I wrote.

Otherwise, this will be set back for some point in the future, and I'll remove it for v0.17.

**Comment by bkazemi** [comment]
I believe I finished it, but there were some bits @sriemer didn't like so I held off on opening a PR. You can have a look on my [code-cleanup](/bkazemi/scanmem/tree/code-cleanup) branch.

**Comment by 12345ieee** [comment]
I had a look, and I have to say I agree with @sriemer , that thing is completely unreadable (I understand it's machine generated, but we've still got to read it all).

Plus, I'm not a fan of touching up a lot of unrelated lines in huge sweep-them-all commits, as it messes up future blames.

I guess for now I'll put this aside, and just change up things that are around where we change code for valid reasons. I like your comment convention, I think it's the one I use as well.

It looks like you fixed typos (in a non machine-generated way, I hope), do you have a commit with only that? That one I'd like to merge.

**Comment by bkazemi** [comment]
Yeah, in retrospect I think the best way to go about it is in individual parts. I have the typo fixes. I'll open a new branch.
